packages/vm/src/imports.rs

```diff
@@ -153,8 +154,10 @@ fn write_to_contract<S: Storage, Q: Querier>(ctx: &mut Ctx, input: &[u8]) -> VmR
 pub fn do_query_chain<S: Storage, Q: Querier>(ctx: &mut Ctx, request_ptr: u32) -> VmResult<u32> {
     let request = read_region(ctx, request_ptr, MAX_LENGTH_QUERY_CHAIN_REQUEST)?;
 
-    let (result, gas_info) =
-        with_querier_from_context::<S, Q, _, _>(ctx, |querier| Ok(querier.raw_query(&request)))?;
+    let gas_remaining = get_gas_left(ctx);
```

Comment:
This is original gas - internal used - externally used (sdk storage), correct?

(I will check the code soon, but if so then looks :100:

Reply:
Yes, looks proper.